src/main/java/org/asia/game/result/JsonFileRepository.java

```
@@ -0,0 +1,112 @@
+package org.asia.game.result;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.asia.game.GameUI;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.asia.game.Config.DIRECTORY_PATH;
+import static org.asia.game.Config.FILE_PATH;
+
+class JsonFileRepository implements GameResultRepository {
+
+    GameUI gameUI = new GameUI();
+
+    @Override
+    public void saveGameResult(GameResult gameResult) {
+
+        var mapper = new ObjectMapper().registerModule(new JavaTimeModule());
+        try {
+            var json = mapper.writeValueAsString(gameResult);
+            gameUI.printSerialisedJson(json);
+            saveGameResultToFile(json);
+        } catch (JsonProcessingException ignored) {}
```

We never should ignore the exception. Please introduce your own runtime exception e.g. `GameReporitoryProcessingException` and throw it whenever you meet an unsolvable condition.

---

Question to line 46: Should I also use my custom exception in this case?